### PR TITLE
feat(always-return): add `ignoreLastCallback` option

### LINKED ISSUE #59: always-return should ignore the last then
Taking the example from the linked article and extending it a bit:

```js
getUserByName('nolan').then(function (user) {
  if (user.isLoggedOut()) {
    throw new Error('user logged out!'); // throwing a synchronous error!
  }
  if (inMemoryCache[user.id]) {
    return inMemoryCache[user.id];       // returning a synchronous value!
  }
  return getUserAccountById(user.id);    // returning a promise!
}).then(function finalThen(userAccount) {
  console.log(userAccount);
}).catch(function (err) {
  console.error(err);
});
```

There's no need to return anything in `finalThen`. Would it be possible to ignore these cases (in the last then, when the promise isn't returned or used in any way)?

The following would become valid:

```js
myPromise.then(function(val) {});
```

The following would remain invalid:

```js
myPromise.then(function(val) { /* missing return here */ }).then(function(val2) { /* no error here */ });
var p = myPromise.then(function(val) {});
return myPromise.then(function(val) {});
```

### PR DESCRIPTION
**Pull request by ota-meshi**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [ ] New rule
- [x] Changes an existing rule
- [ ] Add autofixing to a rule
- [ ] Other, please explain:

**What changes did you make? (Give an overview)**

This PR adds `ignoreLastCallback` option to `always-return` rule.

If `ignoreLastCallback: true`, the last `then()` callback in a promise chain does not warn if it does not have a `return`.

close #59

### CODE REVIEW

File: __tests__/always-return.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const rule = require('../rules/always-return')
 const RuleTester = require('eslint').RuleTester
 const ruleTester = new RuleTester({
   parserOptions: {
-    ecmaVersion: 6,
+    ecmaVersion: 11,
```
Comment (MichaelDeBoey) [inline __tests__/always-return.js:7]: Can we use `ES20XX` here instead?

File: docs/rules/always-return.md [hunk 2]
```diff
@@ -31,3 +39,51 @@ myPromise.then((b) => {
...
 })
 ```
+
+#### Options
+
+##### `ignoreLastCallback`
+
+You can pass an `{ ignoreLastCallback: true }` as an option to this rule to the
+last `then()` callback in a promise chain does not warn if it does not have a
+`return`. Default is `false`.
```
Comment (MichaelDeBoey) [inline docs/rules/always-return.md:49]: I think it would be a good idea to default this to `true` in the next major release

**Review by MichaelDeBoey — APPROVED** [review]

File: docs/rules/always-return.md (hunk 2 above)
Comment (ota-meshi) [reply docs/rules/always-return.md:49]: I also prefer that option to be `true` by default.
But I think the champion's judgment is necessary.

Comment (MichaelDeBoey) [reply docs/rules/always-return.md:49]: Yeah that's why we should do it in a major release, not now

File: __tests__/always-return.js (hunk 1 above)
Comment (MichaelDeBoey) [reply __tests__/always-return.js:7]: I think `ES20XX` makes it more clear as these are the ones that are more known/more used around the community.

Don't think @xjamundx would mind if we change this.

Comment (MichaelDeBoey) [reply __tests__/always-return.js:7]: Sounds like a plan!

**Pull request merged**